bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/Conjunction/Generator.cpp

```diff
@@ -0,0 +1,142 @@
+                    search_tolerance (Duration): The tolerance to use during the close approach search. Defaults to Duration.milliseconds(1.0).
+            )doc",
+            arg("reference_trajectory"),
+            arg("search_step_size") = Duration::Seconds(15.0),
+            arg("search_tolerance") = Duration::Milliseconds(1.0)
+        )
+
+        .def(
+            "is_defined",
+            &Generator::isDefined,
+            R"doc(
+                Check if the generator is defined.
+
+                Returns:
+                    bool: True if generator is defined, False otherwise.
+            )doc"
+        )
+
+        .def(
+            "get_reference_trajectory",
+            &Generator::getReferenceTrajectory,
+            R"doc(
+                Get the reference trajectory.
+
+                Returns:
+                    Trajectory: The reference trajectory.
+            )doc"
+        )
+
+        .def(
+            "get_search_step_size",
+            &Generator::getSearchStepSize,
+            R"doc(
+                Get the search step size.
+
+                Returns:
+                    Duration: The search step size.
+            )doc"
+        )
+
+        .def(
+            "get_search_tolerance",
+            &Generator::getSearchTolerance,
+            R"doc(
+                Get the search tolerance.
+
+                Returns:
+                    Duration: The search tolerance.
+            )doc"
+        )
+
+        .def(
+            "compute_close_approaches",
+            &Generator::computeCloseApproaches,
+            R"doc(
+                Find close approaches between the reference trajectory and another object over a search interval.
+
+                Args:
+                    other_object_trajectory (Trajectory): The trajectory of the other object (Object 2).
+                    search_interval (Interval): The interval over which close approaches are searched.
+
+                Returns:
+                    list[CloseApproach]: Array of close approaches over the search interval (with Object 1 being the reference trajectory).
+            )doc",
+            arg("other_object_trajectory"),
+            arg("search_interval")
+        )
+
+        .def(
+            "set_search_step_size",
+            &Generator::setSearchStepSize,
+            R"doc(
+                Set the search step size.
+
+                Args:
+                    search_step_size (Duration): The search step size.
+            )doc",
+            arg("search_step_size")
+        )
+
+        .def(
+            "set_search_tolerance",
+            &Generator::setSearchTolerance,
+            R"doc(
+                Set the search tolerance.
+
+                Args:
+                    search_tolerance (Duration): The search tolerance.
+            )doc",
+            arg("search_tolerance")
+        )
```

Comment:
```suggestion
```
I'd remove the setters as we typically do not provide setters for trivial types like this, you can just create a new generator.